Userland/Libraries/LibGfx/Bitmap.cpp

```diff
@@ -141,6 +142,19 @@ ErrorOr<NonnullRefPtr<Bitmap>> Bitmap::load_from_file(StringView path, int scale
     return load_from_fd_and_close(fd, path);
 }
 
+ErrorOr<NonnullRefPtr<Bitmap>> Bitmap::load_from_file(NonnullOwnPtr<Core::File> file, StringView path)
+{
+    auto mapped_file = TRY(Core::MappedFile::map_from_file(move(file), path));
+    auto mime_type = Core::guess_mime_type_based_on_filename(path);
+    if (auto decoder = ImageDecoder::try_create_for_raw_bytes(mapped_file->bytes(), mime_type)) {
+        auto frame = TRY(decoder->frame(0));
+        if (auto& bitmap = frame.image)
+            return bitmap.release_nonnull();
+    }
+
+    return Error::from_string_literal("Gfx::Bitmap unable to load from file");
```

Comment:
This is almost identical to `load_from_fd_and_close()`. I don't think that the "file/fd" distinction in the error message is useful enough to need this duplication.

Maybe a `Bitmap::load_from_mapped_file()` could be used for both?

Reply:
`load_from_fd_and_close()` is only used in `load_from_file(StringView)` and 3DFileViewer (this usage is removed later in the PR).

I think we should change `load_from_file(StringView)` and remove `load_from_fd_and_close`. Do you prefer me to do it now or in another PR?

Reply:
Another PR is fine. :^) Let's get this merged!